frontend/packages/console-app/src/components/console-operator/ConsoleOperatorConfig.tsx

```diff
@@ -96,84 +109,117 @@ const ConsolePluginStatus: React.FC<ConsolePluginStatusType> = ({ enabled, plugi
-      const data = pluginInfoEntries.filter(isLoadedDynamicPluginInfo).map((plugin) => {
-        return {
-          name: plugin.metadata.name,
-          version: plugin.metadata.version,
-          description: plugin.metadata?.customProperties?.console?.description || placeholder,
-          enabled: plugin.enabled,
-          status: plugin.status,
-        };
-      });
       setRows(
-        data?.map((item) => {
-          return {
-            cells: [
-              {
-                title: item.name,
-              },
-              item.version,
-              item.description,
-              {
-                title: <Status status={item.status} title={item.status} />,
-              },
-              {
-                title: t('console-app~Enabled'),
-              },
-            ],
-          };
-        }),
+        data.map((item) => ({
+          cells: [
+            {
+              title: item.name,
+            },
+            item.version ?? placeholder,
+            item.description ?? placeholder,
+            {
+              title: (
+                <Status status={item.status} title={item.status}>
+                  {getFailedPluginStatusContent(item)}
+                </Status>
+              ),
+            },
+            {
+              title: item.enabled ? t('console-app~Enabled') : t('console-app~Disabled'),
+            },
+            {
+              title: <ConsolePluginCSPStatus hasViolations={item.hasCSPViolations ?? false} />,
+            },
+          ],
+        })),
       );
       return;
     }
-    const data = consolePlugins.map((plugin) => {
-      const pluginName = plugin?.metadata?.name;
-      const loadedPluginInfo = pluginInfoEntries
-        .filter(isLoadedDynamicPluginInfo)
-        .find(
-          (i: LoadedDynamicPluginInfo) => i?.metadata?.name === pluginName,
-        ) as LoadedDynamicPluginInfo;
-      const notLoadedPluginInfo = pluginInfoEntries
-        .filter(isNotLoadedDynamicPluginInfo)
-        .find(
-          (i: NotLoadedDynamicPluginInfo) => i?.pluginName === pluginName,
-        ) as NotLoadedDynamicPluginInfo;
-      const enabled = !!obj?.spec?.plugins?.includes(pluginName);
-      if (loadedPluginInfo) {
-        return {
-          name: plugin?.metadata?.name,
-          version: loadedPluginInfo?.metadata?.version,
-          description: loadedPluginInfo?.metadata?.customProperties?.console?.description,
-          enabled,
-          status: loadedPluginInfo?.status,
-        };
-      }
-      return {
-        name: plugin?.metadata?.name,
-        enabled,
-        status: notLoadedPluginInfo?.status,
-        errorMessage:
-          notLoadedPluginInfo?.status !== 'Pending' ? notLoadedPluginInfo?.errorMessage : undefined,
-        errorCause:
-          notLoadedPluginInfo?.status !== 'Pending'
-            ? notLoadedPluginInfo?.errorCause?.toString()
-            : undefined,
-      };
+
+    data.forEach((item) => {
```

Comment:
lets use a map there rather, to avoid mutating the item

Reply:
The data (enabled in `PluginStore` :arrow_forward: enabled on cluster) was mutated regardless of the specific implementation.